### PR TITLE
LP-1745 Add unit tests for specialization about view

### PR DESCRIPTION
**Pull request by zeshanarifios**
### Description

[LP-1745](https://philanthropyu.atlassian.net/browse/LP-1745)

### CODE REVIEW

File: openedx/features/specializations/tests/mock_get_program_helpers.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        'marketing_url': 'professional-certificate/discovery-specialization-1',
+        'banner_image': {
+            'x-small': {
+                'url': 'http://localhost:18381/media/media/programs/banner_images/eb228773-a9a5-48cf-bb0e'
+                       '-94725d5aa4f1-c147bfda9f30.x-small.jpg',
+                'width': 348,
+                'height': 116
+            },
+            'large': {
+                'url': 'http://localhost:18381/media/media/programs/banner_images/eb228773-a9a5-48cf-bb0e'
```
[the remark this answers is not in the document]
Comment (zeshanarifios) [reply openedx/features/specializations/tests/mock_get_program_helpers.py:23]: I think it's mocked data, so no need to use dynamic values.

File: openedx/features/specializations/tests/test_views.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    Intercept calls to `render_to_response` and attach the context dict to the
+    response for examination in unit tests.
+    """
+    # I think Django already does this for you in their TestClient, except
+    # we're bypassing that by using edxmako.  Probably edxmako should be
+    # integrated better with Django's rendering and event system.
+    response = render_to_response(path, context)
+    response.mako_context = context
+    response.mako_template = path
+    return response
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply openedx/features/specializations/tests/test_views.py:25]: I have looked into this, view url `response` always contain context, but for some reason it is always `None`.

```
response = self.client.get(self.path)
response.context
```

We will need some configuration to make it work. See [this](https://stackoverflow.com/a/17189587) and [this](https://code.djangoproject.com/ticket/3051#no1)

File: openedx/features/specializations/tests/test_views.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    def test_program_has_no_courses(self, mock_webpack_loader):
+        response = self.client.get(self.path)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.mako_context['courses']), 0)
+
+    @patch('openedx.features.specializations.views.render_to_response', intercept_renderer)
+    @patch.object(DiscoveryClient, '__init__', Mock(return_value=None))
+    @patch.object(DiscoveryClient, 'get_program', Mock(return_value=mock_get_program([mock_course()])))
+    @patch('webpack_loader.loader.WebpackLoader.get_bundle')
+    def test_courses_without_runs(self, mock_webpack_loader):
```
[the remark this answers is not in the document]
Comment (zeshanarifios) [reply openedx/features/specializations/tests/test_views.py:57]: Actually get_programs return python dictionary, So we need dynamic content for courses and its runs in programs to test different scenarios.

File: openedx/features/specializations/tests/test_views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (zeshanarifios) [reply openedx/features/specializations/tests/test_views.py:25]: edx is also using same method for test cases that are using context for assertions. I could not find any other option for this.

**Pull request merged**